src/forms/Slushbucket.tsx

```diff
@@ -0,0 +1,237 @@
+                  return (
+                    <div
+                      tabIndex={i + 1}
+                      key={choiceItem.value + '_' + i}
+                      className={`${styles.ResultRow} ${selected.includes(
+                        choiceItem.value
+                      )
+                        ? styles.Active
+                        : ""}`}
+                      onMouseDown={() => {
+                        if (
+                          selected.filter(e => e === choiceItem.value).length === 0
+                        ) {
+                          selected.push(choiceItem.value);
+                          updateModelValue(selected);
+                        } else {
+                          updateModelValue(
+                            selected.filter(e => e !== choiceItem.value)
+                          );
+                        }
+                      }}
+                    >
+                      {choiceItem.display}
+                    </div>
+                  );
+                })}
+            </Scrollbars>
+          </div>
+          <div
+            className={styles.Results}
+            style={{
+              marginLeft: "20px"
+            }}
+          >
+            <div className={`${styles.SelectedRow}`}>
+              <b>Selected</b>
+            </div>
+            <Scrollbars autoHeight autoHeightMin={400} autoHeightMax={400}>
+              {selected
+                .map((selectedItem) => {
+                  // lookup complete { value, display} object in choices array
+                  return choices.filter(choice => choice.value === selectedItem)[0];
+                })
+                .sort((nameA, nameB) => {
+                  if (nameA.display < nameB.display) {
+                    return -1;
+                  }
+                  if (nameA.display > nameB.display) {
+                    return 1;
+                  }
+                  return 0;
+                })
+                .map((choiceItem,i) => {
+                  return (
+                    <div
+                      className={`${styles.SelectedRow} ${this.props.readonly
+                        ? styles.SelectedRowReadonly
+                        : ""}`}
+                      style={{
+                        display: "flex",
+                        justifyContent: "space-between",
+                        alignItems: "center"
+                      }}
+                      key={choiceItem.value + '_' + i}
+                    >
+                      <div>{choiceItem.display}</div>
+                      <ClearButton
+                        onClick={() => {
+                          updateModelValue(
+                            selected.filter(e => e !== choiceItem.value)
+                          );
+                        }}
+                        style={{
+                          visibility: this.props.readonly ? "hidden" : "visible"
+                        }}
```

Comment:
Strictly speaking this makes the component not readonly, it only hides the button... but that is maggotzifting.

Reply:
This one is already not an input field so otherwise then removing the button with the onclick event I see no need to make it readonly.